include/envoy/http/codec.h

```diff
@@ -189,6 +210,9 @@ struct Http2Settings {
   // our default connection-level window also equals to our stream-level
   static const uint32_t DEFAULT_INITIAL_CONNECTION_WINDOW_SIZE = 256 * 1024 * 1024;
   static const uint32_t MAX_INITIAL_CONNECTION_WINDOW_SIZE = (1U << 31) - 1;
+
+  // By default, do not enforce stream buffer limits beyond the connection level limits.
+  static const uint32_t DEFAULT_PER_STREAM_BUFFER_LIMIT = 0;
```

Comment:
While I'd prefer something smaller in the long run 256M is strictly better than leaving it infinite :-)

I think lowering should be done in a follow-up anyway.   I think there will be complexities we'll want to mull on - @htuch suggests defaulting filter limits to stream limits, which would make this a breaking change for non-streaming filters.